modules/ROOT/pages/apikit-odatav4-expand-feature.adoc

```diff
@@ -0,0 +1,98 @@
+= Implement $expand Feature
+
+`$expand` is a system query option that reduces the amount of incoming network traffic and makes APIs easier to be consumed. Consumers only request inline representations for the related entities of query results so only relevant information is provided avoiding multiple network roundtrips, which is useful for metered and latency-sensitive consumers.
+
+From APIkit for OData 1.3.0, this module supports automatic routing for ease of implementing the `$expand` system query option in most common cases. Though this feature requires more preparation and configuration, it enables the reuse of already implemented flows when processing the `$expand` requests.
+
+For more information on the system query option, see http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionexpand[the OASIS specification^].
+
+== Requirements
+
+- APIkit for OData Module (version 1.3.0).
+- An `<request-entity-collection-listener/>` implementation for each expandable entity.
+- Support for the `in` filter on the `<request-entity-collection-listener/>` flows.
+
+== `$expand` Feature Example 
+
+* The `Employee` entity has a related `Manager` entity.
+* The `Employees` entity set is a collection of `Employee` entities.
+* The `Managers` entity set is a collection of `Manager` entities.
+* API location is `https://example.com/api/`.
+
+Given this setup, requests to retrieve employees with their managers are:
+
+* `GET https://example.com/api/Employees` to retrieve a list of employees.
+* Gather the `Manager` IDs through the `ReferentialConstraint` of their relationship.
+* `GET https://example.com/api/Managers?$filter=ManagerID in (id1, id2, ..., idN)` to retrieve a list of related managers.
+* Write a small program to merge the lists based on join criteria.
+
+If using the `$expand` feature, requests are:
+
+* `GET https://example.com/api/Employees?$expand=Manager` to retrieve a list of employees with their `Manager` navigation property expanded.
+
+== The `generic expand` Feature
+
+The `generic expand` feature offers the possibility to reuse existent flow implementations as a starting point for expanding requests. This feature enables a Mule developer to focus on the relevant data retrieval operations while the module performs the necessary steps to ensure the `$expand` fields get completed.
+
+From the point of view of a Mule developer, an application with the `generic expand` feature enabled works as any other OData Mule application. The difference in the flow execution is that once the information for the main entity set is retrieved, the module will perform additional internal requests gathering the information for any `expand` field. If errors occur during expansion operation, the final behavior is configurable via the error to the consumer, or it can be ignored, resolving it with a `null` value.
```

Comment:
An application with the `generic expand` feature enabled works like any other OData Mule application. However, once the module retrieves information for the main entity set, it performs additional internal requests gathering the information for any `expand` field.

You can write the last sentence more actively. I was not able to do so because I am not sure what you mean by "error to the consumer."

Reply:
I'll take your suggestion and make some changes since I feel some info would be lost otherwise.
And I'll rewrite the "error" part because it is not clear with this wording.
Thanks!